src/Microsoft.Health.Fhir.Core/Features/Search/Converters/FhirTypedElementToSearchValueConverterManager.cs

```diff
@@ -36,5 +41,33 @@ public bool TryGetConverter(string fhirType, Type searchValueType, out ITypedEle
 
             return _converterDictionary.TryGetValue((fhirType, searchValueType), out converter);
         }
+
+        internal class ExtensionConverter<T> : ITypedElementToSearchValueConverter
+        {
+            private readonly IEnumerable<ITypedElementToSearchValueConverter> _underlying;
+            private readonly Type _type;
+
+            public ExtensionConverter(IEnumerable<ITypedElementToSearchValueConverter> underlying)
+            {
+                _underlying = underlying;
+                _type = underlying.First().SearchValueType;
```

Comment:
>underlying.First() [](http://example.com/codeflow?start=24&length=18)

Can we assert Enumerable to have an element?

Reply:
Can we add a comment for whats getting passed in here :)? I'm also not totally following why its a collection but using the first?

Reply:
and do 
```
new ExtensionConverter<int>(group.Key, group.ToList()
```

Reply:
Good idea, implemented this in 51adeb6.